### PR TITLE
Identify sub-modules properly for changelog

### PR DESCRIPTION
**Pull request by zachthompson**
- Allow for an IA to have sub-modules, which will only trigger a "modified" status.
- Skip non-IA updates
- Consolidate error handling

### CODE REVIEW

**Comment by zachthompson** [comment]
@MariagraziaAlastra you'll probably want to be familiar with this since you're doing releases now.  Can be tested with something like this - https://github.com/duckduckgo/zeroclickinfo-goodies/pull/1927

File: lib/Dist/Zilla/Plugin/IAChangelog.pm [hunk 1]
```diff
@@ -73,14 +73,18 @@ sub gather_files {
                 $id = 'cheat_sheets';
             }
             else{
-                if(my $ia = DDG::Meta::Data->get_ia(module => $m)){
-                    unless(@$ia == 1){
-                        $s->log_fatal(["Multiple IDs in metadata for module $m: " . join(', ', map { $_->{id} } @$ia)]); 
+                while($m =~ /::/){
+                    if(my $ia = DDG::Meta::Data->get_ia(module => $m)){
+                        unless(@$ia == 1){
+                            $s->log_fatal(["Multiple IDs in metadata for module $m: " . join(', ', map { $_->{id} } @$ia)]);
```
Comment (MariagraziaAlastra) [inline lib/Dist/Zilla/Plugin/IAChangelog.pm:79]: Wait, so what case would this handle? Do we actually get multiple IDs for a single module?

**Comment by MariagraziaAlastra** [comment]
This LGTM! Ready to merge?

File: lib/Dist/Zilla/Plugin/IAChangelog.pm (hunk 1 above)
Comment (zachthompson) [inline lib/Dist/Zilla/Plugin/IAChangelog.pm:79]: Yes, cheat sheets, for example.  This is to catch cases where the same module name is used or we get another case like cheat sheets we don't know about.

**Comment by zachthompson** [comment]
Yup...merge it.

File: lib/Dist/Zilla/Plugin/IAChangelog.pm (hunk 1 above)
Comment (MariagraziaAlastra) [inline lib/Dist/Zilla/Plugin/IAChangelog.pm:79]: Gotcha

**Pull request merged**